arch/arm/src/stm32h5/hardware/stm32h5xxx_pwr.h

```diff
@@ -0,0 +1,250 @@
+#  define PWR_VMCR_ALS_AVD0     (0 << PWR_VMCR_ALS_SHIFT)
+#  define PWR_VMCR_ALS_AVD1     (1 << PWR_VMCR_ALS_SHIFT)
+#  define PWR_VMCR_ALS_AVD2     (2 << PWR_VMCR_ALS_SHIFT)
+#  define PWR_VMCR_ALS_AVD3     (3 << PWR_VMCR_ALS_SHIFT)
+
+/* USB Supply Control Register */
+
+#define PWR_USBSCR_USB33DEN     (1 << 24) /* Vddusb voltage level detector enable */
+#define PWR_USBSCR_USB33SV      (1 << 25) /* Independent USB Supply Valid */
+
+/* Voltage Monitor Status Register */
+
+#define PWR_VMSR_AVDO           (1 << 19) /* Analog Voltage Detector Output on Vdda */
+#define PWR_VMSR_VDDIO2RDY      (1 << 20) /* Voltage Detector Output on Vddio2 */
+#define PWR_VMSR_PVDO           (1 << 22) /* Programmable Voltage Detect Output */
+#define PWR_VMSR_USB33RDY       (1 << 24) /* Vddusb Ready */
+
+/* Wake-up Status Clear Register */
+
+#define PWR_WUSCR_CWUF1         (1 << 0) /* Clear wake-up pin flag for WUF1 */
+#define PWR_WUSCR_CWUF2         (1 << 1) /* Clear wake-up pin flag for WUF2 */
+#define PWR_WUSCR_CWUF3         (1 << 2) /* Clear wake-up pin flag for WUF3 */
+#define PWR_WUSCR_CWUF4         (1 << 3) /* Clear wake-up pin flag for WUF4 */
+#define PWR_WUSCR_CWUF5         (1 << 4) /* Clear wake-up pin flag for WUF5 */
+#define PWR_WUSCR_CWUF6         (1 << 5) /* Clear wake-up pin flag for WUF6 */
+#define PWR_WUSCR_CWUF7         (1 << 6) /* Clear wake-up pin flag for WUF7 */
+#define PWR_WUSCR_CWUF8         (1 << 7) /* Clear wake-up pin flag for WUF8 */
+
+/* Wake-up Status Register */
+
+#define PWR_WUSR_WUF1           (1 << 0) /* Wake-up event received for WUF1 */
+#define PWR_WUSR_WUF2           (1 << 1) /* Wake-up event received for WUF2 */
+#define PWR_WUSR_WUF3           (1 << 2) /* Wake-up event received for WUF3 */
+#define PWR_WUSR_WUF4           (1 << 3) /* Wake-up event received for WUF4 */
+#define PWR_WUSR_WUF5           (1 << 4) /* Wake-up event received for WUF5 */
+#define PWR_WUSR_WUF6           (1 << 5) /* Wake-up event received for WUF6 */
+#define PWR_WUSR_WUF7           (1 << 6) /* Wake-up event received for WUF7 */
+#define PWR_WUSR_WUF8           (1 << 7) /* Wake-up event received for WUF8 */
+
+/* Wake-up Configuration Register */
+
+#define PWR_WUCR_WUPEN(n)        (1 << (n-1))              /* Enable wake-up pin WUPn. n = 1..8 */
+#define PWR_WUCR_WUPP(n)         (1 << (n+7))              /* Wake-up Pin Polarity bit for WUPn */
+#define PWR_WUCR_MASK_WUPPUPD(n) (3 << (((n-1) * 2) + 16)) /* Wake-up pin pull configuration */
+#define PWR_WUCR_NOPU_WUPPUPD(n) (0 << (((n-1) * 2) + 16))
+#define PWR_WUCR_PU_WUPPUPD(n)   (1 << (((n-1) * 2) + 16))
+#define PWR_WUCR_PD_WUPPUPD(n)   (2 << (((n-1) * 2) + 16))
+
+/* IO Retention Register */
+
+#define PWR_IORETR_IORETEN       (1 << 0)  /* IO Retention Enable */
+#define PWR_IORETR_JTAGIORETEN   (1 << 16) /* IO Retention Enable for JTAG IOs */
+
+/* Power secure configuration register */
+
+#define PWR_SECCFGR_WUP(n)SEC    (1 << (n-1)) /* WUP(n) Secure Protection */
```

Comment:
Is this legal preprocessor syntax? I have never seen that before. I cannot seem to find a reference in GNU C Manual or elsewhere. Normally the parameter will be at the end of the macro name. For example:

```suggestion
#define PWR_SECCFGR_WUP_SEC(n)    (1 << (n-1)) /* WUP(n) Secure Protection */
```

I recommend to avoid syntax which might work with one compiler and fail with another.

Reply:
That was a typo, not intentional. Thank you for catching that, I will update accordingly!